File: backend/src/main/java/dev/k2/commons/FileDescription.java
package dev.k2.commons;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@NoArgsConstructor
public class FileDescription {

    @Id
    @GeneratedValue
    private Long id;

    private String originalFilename;

    @Column(unique = true)
    private String generatedFilename;

}

File: frontend/src/app/components/file-description.ts
export class FileDescription {
  id: number;
  originalFilename: string;
  generatedFilename: string;

  constructor(id: number, originalFilename: string, generatedFilename: string) {
    this.id = id;
    this.originalFilename = originalFilename;
    this.generatedFilename = generatedFilename;
  }
}
